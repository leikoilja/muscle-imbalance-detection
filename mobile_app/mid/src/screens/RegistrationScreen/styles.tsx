import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  descText: {},
  input: {
    borderRadius: 5,
    overflow: 'hidden',
    marginTop: 10,
    marginBottom: 10,
    width: '90%',
  },
});
