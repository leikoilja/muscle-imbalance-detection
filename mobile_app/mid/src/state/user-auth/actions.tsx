import { firebase } from "../../firebase/config";
import {
  LogoutFinishedAction,
  User,
  LoginStartAction,
  LoginFinishedAction,
  LoginErrorAction,
  LoginUserAction,
  LogoutUserAction,
  UserAuthAction,
  LogoutStartAction,
  LogoutErrorAction,
  RegistrationErrorAction,
  RegistrationStartAction,
  RegistrationFinishedAction,
  RegisterUserAction,
} from "../types";
import { Dispatch } from "redux";

export enum USER_AUTH_ACTION_TYPES {
  LOGIN_START = "USER_AUTH/LOGIN_START",
  LOGIN_FINISHED = "USER_AUTH/LOGIN_FINISHED",
  LOGIN_ERROR = "USER_AUTH/LOGIN_ERROR",
  LOGOUT_START = "USER_AUTH/LOGOUT_START",
  LOGOUT_FINISHED = "USER_AUTH/LOGOUT_FINISHED",
  LOGOUT_ERROR = "USER_AUTH/LOGOUT_ERROR",
  REGISTRATION_START = "USER_AUTH/REGISTRATION_START",
  REGISTRATION_FINISHED = "USER_AUTH/REGISTRATION_FINISHED",
  REGISTRATION_ERROR = "USER_AUTH/REGISTRATION_ERROR",
  SAVE_MEASUREMENT = "USER/SAVE_MEASUREMENT",
}

// Login
export const loginStart = (): LoginStartAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGIN_START,
});

export const loginFinished = (user: User): LoginFinishedAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGIN_FINISHED,
  user,
});

export const loginError = (error: string): LoginErrorAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGIN_ERROR,
  error,
});

// Logout
export const logoutStart = (): LogoutStartAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGOUT_START,
});

export const logoutFinished = (): LogoutFinishedAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGOUT_FINISHED,
});

export const logoutError = (error: string): LogoutErrorAction => ({
  type: USER_AUTH_ACTION_TYPES.LOGOUT_ERROR,
  error,
});

// Registration
export const registrationStart = (): RegistrationStartAction => ({
  type: USER_AUTH_ACTION_TYPES.REGISTRATION_START,
});

export const registrationFinished = (
  user: User
): RegistrationFinishedAction => ({
  type: USER_AUTH_ACTION_TYPES.REGISTRATION_FINISHED,
  user,
});

export const registrationError = (error: string): RegistrationErrorAction => ({
  type: USER_AUTH_ACTION_TYPES.REGISTRATION_ERROR,
  error,
});

// User related
export const saveMeasurement = (measurementUid: string) => ({
  type: USER_AUTH_ACTION_TYPES.SAVE_MEASUREMENT,
  measurementUid,
});

//
// Action creators
//
export const loginUser = (
  email: string,
  password: string
): LoginUserAction => async (dispatch) => {
  dispatch(loginStart());
  let user_data = {};
  await firebase
    .auth()
    .signInWithEmailAndPassword(email, password)
    .then((response) => {
      let user = response.user;
      firebase
        .firestore()
        .collection("users")
        .doc(user.uid)
        .get()
        .then((doc) => {
          if (!doc.exists) {
            console.log("No documents");
          } else {
            user_data = doc.data();
          }
        })
        .catch((error) => {
          dispatch(loginError(error));
        })
        .finally(() => {
          user = { ...user_data, user };
          dispatch(loginFinished(user));
        });
    })
    .catch((error) => {
      dispatch(loginError(error));
    });
};

export const logoutUser = (): LogoutUserAction => async (
  dispatch: Dispatch<UserAuthAction>
) => {
  dispatch(logoutStart());
  try {
    await firebase.auth().signOut();
    dispatch(logoutFinished());
  } catch (error) {
    dispatch(logoutError(error));
  }
};

export const registerUser = (
  email: string,
  password: string,
  fullName: string
): RegisterUserAction => async (dispatch) => {
  dispatch(registrationStart());
  await firebase
    .auth()
    .createUserWithEmailAndPassword(email, password)
    .then((response) => {
      const user = response.user;
      const account = {
        fullName: fullName,
        isDoctor: false,
        isSuperuser: false,
      };
      firebase.firestore().collection("users").doc(user.uid).set(account);
      dispatch(registrationFinished(user));
    })
    .catch((error) => {
      dispatch(registrationError(error));
    });
};
