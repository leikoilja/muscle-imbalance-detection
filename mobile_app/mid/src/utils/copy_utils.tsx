export function deepCopy(src) {
  let target = Array.isArray(src) ? [] : {};
  for (let key in src) {
    let v = src[key];
    if (v) {
      if (typeof v === "object") {
        target[key] = deepCopy(v);
      } else {
        target[key] = v;
      }
    } else {
      target[key] = v;
    }
  }

  return target;
}
