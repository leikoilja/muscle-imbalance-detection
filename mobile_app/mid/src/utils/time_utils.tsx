import moment from "moment";

const pad2 = (number) => (number < 10 ? "0" : "") + number;

const baseConversion = (unixTimestamp) => {
  const a = new Date(unixTimestamp * 1000);
  const months = [
    "Jan",
    "Feb",
    "Mar",
    "Apr",
    "May",
    "Jun",
    "Jul",
    "Aug",
    "Sep",
    "Oct",
    "Nov",
    "Dec",
  ];
  const year = a.getFullYear();
  const month = months[a.getMonth()];
  const date = pad2(a.getDate());
  const hour = pad2(a.getHours());
  const min = pad2(a.getMinutes());
  const sec = pad2(a.getSeconds());
  const time = `${date} ${month} ${year} ${hour}:${min}:${sec}`;
  return time;
};

export function unixTimestampToDate(unixTimestamp) {
  return baseConversion(unixTimestamp);
}

export function unixTimestampToDateNoSeconds(unixTimestamp) {
  const time = baseConversion(unixTimestamp);
  return time.slice(0, time.length - 3);
}

export function unixTimestampToDateNoDate(unixTimestamp) {
  const time = baseConversion(unixTimestamp);
  return time.slice(12, time.length);
}

const timeFormat = "YYYY-MM-DD HH:mm";

export function ISOtimestampToString(ISOtimestamp) {
  return moment(ISOtimestamp).format(timeFormat);
}
