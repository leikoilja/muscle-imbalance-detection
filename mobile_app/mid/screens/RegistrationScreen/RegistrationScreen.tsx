import React, { useState } from "react";
import { Image, Text, TextInput, TouchableOpacity, View } from "react-native";
import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
import { firebase } from "../../firebase/config";
import styles from "./styles";

export default function RegistrationScreen({ navigation }) {
  const [fullName, setFullName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");

  const onFooterLinkPress = () => {
    navigation.navigate("Login");
  };

  const onRegisterPress = () => {
    if (password !== confirmPassword) {
      alert("Passwords don't match.");
      return;
    }
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then((response) => {
        const uid = response.user.uid;
        const userData = {
          id: uid,
          email,
          fullName,
        };
        const usersRef = firebase.firestore().collection("users"); // TODO: Fix firebase DB permissions
        usersRef
          .doc(uid)
          .set(userData)
          .then(() => {
            navigation.navigate("Main", { user: userData });
          })
          .catch((error) => {
            alert(error);
          });
      })
      .catch((error) => {
        alert(error);
      });
  };

  return (
    <View style={styles.container}>
      <KeyboardAwareScrollView
        style={{ flex: 1, width: "100%" }}
        keyboardShouldPersistTaps="always"
      >
        <Image
          style={styles.logo}
          source={require("../../assets/images/logo.png")}
        />
        <TextInput
          style={styles.input}
          placeholder="Full Name"
          placeholderTextColor="#aaaaaa"
          onChangeText={(text) => setFullName(text)}
          value={fullName}
          underlineColorAndroid="transparent"
          autoCapitalize="none"
        />
        <TextInput
          style={styles.input}
          placeholder="E-mail"
          placeholderTextColor="#aaaaaa"
          onChangeText={(text) => setEmail(text)}
          value={email}
          underlineColorAndroid="transparent"
          autoCapitalize="none"
        />
        <TextInput
          style={styles.input}
          placeholderTextColor="#aaaaaa"
          secureTextEntry
          placeholder="Password"
          onChangeText={(text) => setPassword(text)}
          value={password}
          underlineColorAndroid="transparent"
          autoCapitalize="none"
        />
        <TextInput
          style={styles.input}
          placeholderTextColor="#aaaaaa"
          secureTextEntry
          placeholder="Confirm Password"
          onChangeText={(text) => setConfirmPassword(text)}
          value={confirmPassword}
          underlineColorAndroid="transparent"
          autoCapitalize="none"
        />
        <TouchableOpacity
          style={styles.button}
          onPress={() => onRegisterPress()}
        >
          <Text style={styles.buttonTitle}>Create account</Text>
        </TouchableOpacity>
        <View style={styles.footerView}>
          <Text style={styles.footerText}>
            Already got an account?{" "}
            <Text onPress={onFooterLinkPress} style={styles.footerLink}>
              Log in
            </Text>
          </Text>
        </View>
      </KeyboardAwareScrollView>
    </View>
  );
}
