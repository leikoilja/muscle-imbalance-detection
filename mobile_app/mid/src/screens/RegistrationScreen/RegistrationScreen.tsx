import React, { useState, Component } from "react";
import { Image, Text, TextInput, TouchableOpacity, View } from "react-native";
import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
import { registerUser } from "../../state/user-auth/actions";
import styles from "./styles";
import { connect } from "react-redux";
import Button from "../../components/Button";

class RegistrationScreen extends Component {
  state = {
    fullNameValue: "",
    emailValue: "",
    passwordValue: "",
    confirmPasswordValue: "",
  };

  onChange = (e, type) => {
    this.setState({ [`${type}Value`]: e });
  };

  onFooterLinkPress = () => {
    const { navigation } = this.props;
    navigation.navigate("Login");
  };

  onRegisterPress = async () => {
    const { registerUser } = this.props;
    await registerUser(
      this.state.emailValue,
      this.state.passwordValue,
      this.state.fullNameValue
    );
  };

  render() {
    const {
      emailValue,
      passwordValue,
      confirmPasswordValue,
      fullNameValue,
    } = this.state;
    return (
      <View style={styles.container}>
        <KeyboardAwareScrollView
          style={{ flex: 1, width: "100%" }}
          keyboardShouldPersistTaps="always"
        >
          <Image
            style={styles.logo}
            testID="form-logo"
            source={require("../../assets/images/logo.png")}
          />
          <TextInput
            testID="form-fullName-input"
            style={styles.input}
            placeholder="Full Name"
            placeholderTextColor="#aaaaaa"
            onChangeText={(text) => this.onChange(text, "fullName")}
            value={fullNameValue}
            underlineColorAndroid="transparent"
            autoCapitalize="none"
          />
          <TextInput
            testID="form-email-input"
            style={styles.input}
            placeholder="E-mail"
            placeholderTextColor="#aaaaaa"
            onChangeText={(text) => this.onChange(text, "email")}
            value={emailValue}
            underlineColorAndroid="transparent"
            autoCapitalize="none"
          />
          <TextInput
            testID="form-password-input"
            style={styles.input}
            placeholderTextColor="#aaaaaa"
            secureTextEntry
            placeholder="Password"
            onChangeText={(text) => this.onChange(text, "password")}
            value={passwordValue}
            underlineColorAndroid="transparent"
            autoCapitalize="none"
          />
          <TextInput
            testID="form-confirmPassword-input"
            style={styles.input}
            placeholderTextColor="#aaaaaa"
            secureTextEntry
            placeholder="Confirm Password"
            onChangeText={(text) => this.onChange(text, "confirmPassword")}
            value={confirmPasswordValue}
            underlineColorAndroid="transparent"
            autoCapitalize="none"
          />
          <Button
            testID="form-registration-button"
            onPress={this.onRegisterPress}
            text="Create account"
          />
          <View style={styles.footerView}>
            <Text style={styles.footerText}>
              Already got an account?{" "}
              <Text
                testID="form-footer-button"
                onPress={this.onFooterLinkPress}
                style={styles.footerLink}
              >
                Log in
              </Text>
            </Text>
          </View>
        </KeyboardAwareScrollView>
      </View>
    );
  }
}

const mapStateToProps = (state) => ({
  auth: state.userAuth,
});

const mapDispatchToProps = {
  registerUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(RegistrationScreen);
