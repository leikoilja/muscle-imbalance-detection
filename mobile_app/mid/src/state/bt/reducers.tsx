import { BT_ACTION_TYPES } from "./actions";

const initialState = {
  device: {},
  isConnected: false,
};

export const btReducer = (state = initialState, action) => {
  switch (action.type) {
    case BT_ACTION_TYPES.SAVE_DEVICE:
      const { device } = action;
      return {
        ...state,
        device,
      };
    case BT_ACTION_TYPES.UPDATE_DEVICE_STATE:
      const { isConnected } = action;
      return {
        ...state,
        isConnected,
      };
    default:
      return state;
  }
};
