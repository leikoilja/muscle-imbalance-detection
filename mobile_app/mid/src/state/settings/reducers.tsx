import { SETTINGS_ACTION_TYPES } from "./actions";

const initialState = {
  bt: {
    device: {},
    isConnected: false,
  },
  theme: "light",
};

export const settingsReducer = (state = initialState, action) => {
  switch (action.type) {
    case SETTINGS_ACTION_TYPES.BT_SAVE_DEVICE:
      const { device } = action;
      return {
        ...state,
        bt: { ...state.bt, device },
      };
    case SETTINGS_ACTION_TYPES.BT_REMOVE_DEVICE:
      return {
        ...state,
        bt: initialState.bt,
      };
    case SETTINGS_ACTION_TYPES.BT_UPDATE_DEVICE_STATE:
      const { isConnected } = action;
      return {
        ...state,
        isConnected,
      };
    case SETTINGS_ACTION_TYPES.THEME_CHANGE:
      const { theme } = action;
      return {
        ...state,
        theme,
      };
    default:
      return state;
  }
};
