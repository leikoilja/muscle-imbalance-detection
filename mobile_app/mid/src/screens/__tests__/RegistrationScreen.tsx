import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { setMockStore } from "../../utils/test_utils";
import RegistrationScreen from "../RegistrationScreen/RegistrationScreen";
import { initialState } from "../../state/user-auth/reducers";

const createTestProps = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

describe("RegistrationScreen", () => {
  let store;
  let component;
  let props;

  beforeEach(() => {
    props = createTestProps({});
    [store, component] = setMockStore(
      { userAuth: initialState },
      <RegistrationScreen {...props} />
    );
    store.dispatch = jest.fn();
  });

  it("shows registration form", () => {
    const { getByTestId, toJSON } = render(component);
    expect(getByTestId("form-logo")).toBeTruthy();
    expect(getByTestId("form-fullName-input")).toBeTruthy();
    expect(getByTestId("form-email-input")).toBeTruthy();
    expect(getByTestId("form-password-input")).toBeTruthy();
    expect(getByTestId("form-confirmPassword-input")).toBeTruthy();
    expect(getByTestId("form-registration-button")).toBeTruthy();
    expect(getByTestId("form-footer-button")).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it("navigates to Login if footer link is pressed", () => {
    const { getByTestId } = render(component);
    const button = getByTestId("form-footer-button");
    fireEvent(button, "press");
    expect(props.navigation.navigate).toHaveBeenCalledTimes(1);
    expect(props.navigation.navigate).toHaveBeenCalledWith("Login");
  });

  it("changes emailValue when typing", () => {
    const { getByTestId } = render(component);
    const textInput = getByTestId("form-email-input");
    fireEvent.changeText(textInput, "example@company.com");
    expect(textInput.props.value).toBe("example@company.com");
  });

  it("changes passwordValue when typing", () => {
    const { getByTestId } = render(component);
    const textInput = getByTestId("form-password-input");
    fireEvent.changeText(textInput, "password");
    expect(textInput.props.value).toBe("password");
  });

  it("loggs-in when Login button is pressed", () => {
    const { getByTestId } = render(component);
    const button = getByTestId("form-registration-button");
    fireEvent(button, "press");
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    // TODO: Check that registerUser dispatch has been called
  });
});
