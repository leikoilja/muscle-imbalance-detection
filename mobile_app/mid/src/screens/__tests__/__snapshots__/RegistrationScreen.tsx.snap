// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationScreen shows registration form 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="always"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "flex": 1,
            "height": 120,
            "margin": 30,
            "resizeMode": "contain",
            "width": "85%",
          }
        }
        testID="form-logo"
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Full Name"
        placeholderTextColor="#aaaaaa"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#f5f5f5",
            "borderRadius": 5,
            "height": 48,
            "marginBottom": 10,
            "marginLeft": 30,
            "marginRight": 30,
            "marginTop": 10,
            "overflow": "hidden",
            "paddingLeft": 16,
          }
        }
        testID="form-fullName-input"
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="E-mail"
        placeholderTextColor="#aaaaaa"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#f5f5f5",
            "borderRadius": 5,
            "height": 48,
            "marginBottom": 10,
            "marginLeft": 30,
            "marginRight": 30,
            "marginTop": 10,
            "overflow": "hidden",
            "paddingLeft": 16,
          }
        }
        testID="form-email-input"
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#aaaaaa"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#f5f5f5",
            "borderRadius": 5,
            "height": 48,
            "marginBottom": 10,
            "marginLeft": 30,
            "marginRight": 30,
            "marginTop": 10,
            "overflow": "hidden",
            "paddingLeft": 16,
          }
        }
        testID="form-password-input"
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Confirm Password"
        placeholderTextColor="#aaaaaa"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#f5f5f5",
            "borderRadius": 5,
            "height": 48,
            "marginBottom": 10,
            "marginLeft": 30,
            "marginRight": 30,
            "marginTop": 10,
            "overflow": "hidden",
            "paddingLeft": 16,
          }
        }
        testID="form-confirmPassword-input"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#788eec",
            "borderRadius": 5,
            "height": 48,
            "justifyContent": "center",
            "marginLeft": 30,
            "marginRight": 30,
            "marginTop": 20,
            "opacity": 1,
          }
        }
        testID="form-registration-button"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Create account
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "marginTop": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2e2e2d",
              "fontSize": 16,
            }
          }
        >
          Already got an account?
           
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#788eec",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
            testID="form-footer-button"
          >
            Log in
          </Text>
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
